src/mcp_ynab/server.py

```
@@ -592,17 +599,35 @@ async def categorize_transaction(
-        response = transactions_api.get_transactions(budget_id, since_date=since_date)
-        target_transaction = _find_transaction_by_id(
-            response.data.transactions, transaction_id, id_type
-        )
+        target_transaction: Optional[TransactionDetail] = None
+        if id_type == "id":
+            try:
+                single_response = transactions_api.get_transaction_by_id(
+                    budget_id=budget_id, transaction_id=transaction_id
+                )
+                target_transaction = single_response.data.transaction
+            except Exception:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)